mu-plugins/pwcc-helpers/inc/tachyon-mods/namespace.php

```diff
@@ -0,0 +1,315 @@
+ * @param $image array The image details.
+ *
+ * @return array Modified arguments with gravity corrected.
+ */
+function filter_tachyon_gravity( $tachyon_args, $image ) {
+	if ( ! is_array( $tachyon_args ) || ! isset( $tachyon_args['gravity'] ) ) {
+		return $tachyon_args;
+	}
+
+	$gravity = $tachyon_args['gravity'];
+
+	// Ensure Gravity is not empty.
+	if ( $gravity === '' ) {
+		unset( $tachyon_args['gravity'] );
+		return $tachyon_args;
+	}
+
+	// Check if too short to need modification.
+	if ( strlen( $gravity ) <= 5 ) {
+		return $tachyon_args;
+	}
+
+	// Ensure direction is correct: northeast instead of eastnorth.
+	$gravity_tail = substr( $gravity, -5 );
+	if ( $gravity_tail === 'north' || $gravity_tail === 'south' ) {
+		$gravity = $gravity_tail . substr( $gravity, 0, -5 );
+	}
+
+	// Normalise for orientation.
+	$image_id  = $image['attachment_id'];
+	$image_meta  = wp_get_attachment_metadata( $image_id );
+
+	$orientation = '';
+	if ( isset( $image_meta['height'], $image_meta['width'] ) ) {
+		$orientation = ( $image_meta['height'] > $image_meta['width'] ) ? 'portrait' : 'landscape';
+	}
+
+	switch ( $orientation ) {
+		case 'portrait':
+			if ( strpos( $gravity, 'north' ) !== false ) {
+				$gravity = 'north';
+			} elseif ( strpos( $gravity, 'south' ) !== false ) {
+				$gravity = 'south';
+			}
+			break;
+		case 'landscape':
+			if ( strpos( $gravity, 'east' ) !== false ) {
+				$gravity = 'east';
+			} elseif ( strpos( $gravity, 'west' ) !== false ) {
+				$gravity = 'west';
+			}
+			break;
+	}
+
+	$tachyon_args['gravity'] = $gravity;
+	return $tachyon_args;
+}
+
+/**
+ * Filter the meta data WordPress uses to generate the srcset.
+ *
+ * @TODO Work out how the h*ck to fix this.
+ *
+ * @param array  $image_meta    The image meta data as returned by 'wp_get_attachment_metadata()'.
+ * @param array  $size_array    Array of width and height values in pixels (in that order).
+ * @param string $image_src     The 'src' of the image.
+ * @param int    $attachment_id The image attachment ID or 0 if not supplied.
+ *
+ * @return array The modified image meta for generating the srcset.
+ */
+function filter_image_srcset_meta( $image_meta, $size_array, $image_src, $attachment_id ) {
+	/*
+	 * Because I use Tachyon in the admin/inserted data at this point
+	 * WordPress is very confused and returns the full size image as the URL.
+	 *
+	 * This in turn confuses Tachyon so it doesn't know which crop to use either.
+	 *
+	 * The result of all this confusing is that the wrong size images get included
+	 * in the srcset. Which is a problem.
+	 *
+	 * So I've ripped out the srcset as a temporary measure.
+	 *
+	 * CAUSE
+	 *
+	 * In `wp_image_add_srcset_and_sizes()` WordPress removes the querystring
+	 * from any WordPress image URLs.
+	 *
+	 * In this filter `$image_src` becomes incorrect as a result:
+	 * this: <TACHYON_URL>/image.ext?resize=400,400&crop=north
+	 * becomes: <TACHYON_URL>/image.ext
+	 *
+	 * POTENTIAL FIXES
```

Comment:
Take the second approach:

Replace the `wp_make_content_images_responsive` filter on `the_content`, that seems less expensive but will be a pain to maintain. It require some sort of forking to deal with old posts.